src/components/EmojiPicker.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useCallback, useRef, useState } from 'react';
+import { EmojiData, Picker } from 'emoji-mart';
+
+import { useTranslationContext } from '../Context';
+import { useOnClickOutside } from '../hooks/useOnClickOutside';
+
+export type EmojiPickerType = {
+  onSelect?: (emoji: EmojiData) => void;
+};
+
+export const EmojiPicker = ({ onSelect }: EmojiPickerType) => {
+  const { t } = useTranslationContext();
+  const [open, setOpen] = useState(false);
+  const emojiPicker = useRef<HTMLDivElement | null>(null);
+
+  const closeMenu = useCallback(() => setOpen(false), []);
+  useOnClickOutside(emojiPicker, closeMenu);
```

Comment:
```suggestion
  useOnClickOutside(emojiPicker, closeMenu, open);
```

Reply:
If `open` is set to true, events get registered and after user clicks somewhere outside actual referenced container the cleaner gets called before new effect and listeners get removed and since the condition is not true anymore, new listeners get registered next time open is set to `true`.

Reply:
good point, pushed a fix 👍